jetty-core/jetty-core-server/src/main/java/org/eclipse/jetty/core/server/Request.java

```diff
@@ -293,22 +299,22 @@ public Request getWrapped()
         }
     }
 
-    abstract class ProcessingWrapper extends Wrapper implements Handler.Processor
+    abstract class ProcessingWrapper extends Wrapper implements Processor
```

Comment:
By all the other naming, this is a `WrapperProcessor`.

Reply:
This class is primarily a request Wrapper, that you can call `asProcessor` on to get some utility logic to provide a Processor wrapper that substitutes the request in the processing chain.  It is an implementation detail that this class is-a Processor and it could be implemented with an internal private (or dynamic) processor.... but that would avoid the benefit of creating only 1 object.   The class is only used as a Wrapper and never directly used as a Processor.

So I'm happy to consider better names, but it is-not-a Processor.

Reply:
If the process was a final field within the wrapper class, would the optimizer treat an allocation of the wrapper as 1 object or 2?